src/pages/Assets/Overview/YourSynths/YourSynths.tsx

```diff
@@ -80,6 +83,29 @@ export const YourSynths: FC<YourSynthsProps> = memo(
 			setSelectedSynth(null);
 		};
 
+		const handleAddToMetamask = async (currencyKey: string) => {
+			try {
+				const tokenDetails = await getCurrencyDetails(currencyKey);
```

Comment:
To get the correct address (for all networks), we use `snxJSConnector` (you can find multiple usages of it). 

So you can do something like

```
const { snxJS } = snxJSConnector as any; // typing is not really good for snxJS for now.

const contract = snxJS[currencyKey].contract;
// from here you can get everything you need
```